src/vm/ilmarshalers.cpp

```diff
@@ -3807,9 +3803,13 @@ void ILNativeArrayMarshaler::EmitMarshalArgumentCLRToNative()
         EmitLoadManagedValue(m_pcsMarshal);
         m_pcsMarshal->EmitSTLOC(dwPinnedLocal);
         m_pcsMarshal->EmitLDLOC(dwPinnedLocal);
+        m_pcsMarshal->EmitLDLEN();
```

Comment:
I think most of this is straight forward C#, but I really like the fact that much of this IL has comments in C#.